benchmark/pkg/runselector/schemes/last_n_hours.go

```diff
@@ -0,0 +1,49 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schemes
+
+import (
+	"time"
+
+	"k8s.io/contrib/test-utils/utils"
+)
+
+// Returns a list of run numbers of 'job' from last 'n' hours, that successfully
+// completed. If it cannot find n runs, returns as many runs as it could collect.
+// Note: We check if start time of the run is not earlier than 'n' hours from now.
+func GetJobRunsFromLastNHours(job string, n int) ([]int, error) {
+	googleGCSBucketUtils := utils.NewUtils(utils.KubekinsBucket, utils.LogDir)
+	latestRunNumber, err := googleGCSBucketUtils.GetLastestBuildNumberFromJenkinsGoogleBucket(job)
+	if err != nil {
+		return nil, err
+	}
+
+	var runs []int
+	currentTime := uint64(time.Now().Unix())
+	for runNumber := latestRunNumber; runNumber > 0; runNumber-- {
+		if startStatus, err := googleGCSBucketUtils.CheckStartedStatus(job, runNumber); err == nil {
+			if (currentTime - startStatus.Timestamp) < uint64(3600 * n) {
```

Comment:
But I guess timestamp is absolute, right? Something like the number of seconds elapsed since the start of the universe (iirc 00:00:00 UTC, 1st Jan 1970)